elasticsearch_dsl/field.py

```diff
@@ -282,39 +287,74 @@ def clean(self, data):
             raise ValidationException("Value required for this field.")
         return data
 
+
 class Float(Field):
     name = 'float'
 
-class HalfFloat(Field):
+    def _deserialize(self, data):
+        if data is None:
+            return None
+        return float(data)
+
+class HalfFloat(Float):
     name = 'half_float'
 
-class ScaledFloat(Field):
+class ScaledFloat(Float):
     name = 'scaled_float'
 
     def __init__(self, scaling_factor, *args, **kwargs):
         super(ScaledFloat, self).__init__(scaling_factor=scaling_factor, *args, **kwargs)
 
 
-class Double(Field):
+class Double(Float):
     name = 'double'
 
-class Byte(Field):
-    name = 'byte'
-
-class Short(Field):
-    name = 'short'
 
 class Integer(Field):
     name = 'integer'
 
-class Long(Field):
+    def _deserialize(self, data):
+        if data is None:
+            return None
+        return int(data)
+
+
+class Byte(Integer):
+    name = 'byte'
+
+class Short(Integer):
+    name = 'short'
+
+class Long(Integer):
     name = 'long'
 
 class Ip(Field):
     name = 'ip'
 
-class Attachment(Field):
-    name = 'attachment'
+    def _deserialize(self, data):
+        if data is None:
+            return None
+
+        # the ipaddress library for pypy, python2.5 and 2.6 only accepts unicode.
+        return ipaddress.ip_interface(six.u(data))
```

Comment:
We should be using `ip_address` instead of `ip_interface` here since that's the only supported value by elasticsearch - the `ip` field doesn't support ranges or blocks.

Reply:
oh right, it's the search API which should use an `ip_interface`.